spec/index.bs

```diff
@@ -5102,76 +5110,81 @@ A {{GPUBindGroupLayout}} object has the following internal slots:
-                                    - |entry|.{{GPUBindGroupLayoutEntry/buffer}}?.{{GPUBufferBindingLayout/type}}
-                                        must not be {{GPUBufferBindingType/"storage"}}.
-                                    - |entry|.{{GPUBindGroupLayoutEntry/storageTexture}}?.{{GPUStorageTextureBindingLayout/access}}
-                                        must not be {{GPUStorageTextureAccess/"write-only"}}.
-
-                                - If the |textureLayout| is not `undefined` and
-                                    |textureLayout|.{{GPUTextureBindingLayout/multisampled}} is `true`:
-                                    - |textureLayout|.{{GPUTextureBindingLayout/viewDimension}} is
-                                        {{GPUTextureViewDimension/"2d"}}.
-                                    - |textureLayout|.{{GPUTextureBindingLayout/sampleType}} is not
-                                        {{GPUTextureSampleType/"float"}}.
-
-                                - If |storageTextureLayout| is not `undefined`:
-                                    - |storageTextureLayout|.{{GPUStorageTextureBindingLayout/viewDimension}} is not
-                                        {{GPUTextureViewDimension/"cube"}} or {{GPUTextureViewDimension/"cube-array"}}.
-                                    - |storageTextureLayout|.{{GPUStorageTextureBindingLayout/format}} must be a format
-                                        which can support storage usage.
-                        </div>
+                1. If any of the following conditions are unsatisfied
+                    [$generate a validation error$], make |layout| [=invalid=], and stop.
 
-                    1. Set |layout|.{{GPUBindGroupLayout/[[descriptor]]}} to |descriptor|.
-                    1. Set |layout|.{{GPUBindGroupLayout/[[dynamicOffsetCount]]}} to the number of
-                        entries in |descriptor| where {{GPUBindGroupLayoutEntry/buffer}} is not `undefined` and
-                        {{GPUBindGroupLayoutEntry/buffer}}.{{GPUBufferBindingLayout/hasDynamicOffset}} is `true`.
-                    1. For each {{GPUBindGroupLayoutEntry}} |entry| in
-                        |descriptor|.{{GPUBindGroupLayoutDescriptor/entries}}:
-                        1. Insert |entry| into |layout|.{{GPUBindGroupLayout/[[entryMap]]}}
-                            with the key of |entry|.{{GPUBindGroupLayoutEntry/binding}}.
-                </div>
-            1. Return |layout|.
+                    <div class=validusage>
+                        - |this| is [=valid=].
+                        - Let |limits| be |this|.{{GPUObjectBase/[[device]]}}.{{device/[[limits]]}}.
+                        - The {{GPUBindGroupLayoutEntry/binding}} of each entry in |descriptor| is unique.
+                        - The {{GPUBindGroupLayoutEntry/binding}} of each entry in |descriptor| must be &lt;
+                            |limits|.{{supported limits/maxBindingIndex}}.
```

Comment:
the name `maxBindingIndex` makes it sound like it should be a "&le;". "&le" is probably fine, I don't have a good alternative name suggestion.

Reply:
I don't have a great name suggestion either that would allow it to continue being a "<". It's a little weird for it to be a "≤", since that means technically we would allow 641 unique binding indices, but given that it's something of an in-joke anyway I don't think that matters. In the future if we have a "real" limit it won't look that weird to have some square number-1, as most devs will recognize where that comes from. (Plus we use similar verbiage for things like `maxComputeWorkgroupSize*` so it's not completely without precedent.)

Reply:
If we really wanted it to be 640 we could maybe name it `maxBindingsPerBindGroup`, it would just be slightly misleading because you can never actually have that many bindings per bind group (other limits are the limiting factor). I'm fine with 641. *shrug*